plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/AwsS3Service.java

```diff
@@ -19,151 +19,156 @@
 
 package org.elasticsearch.cloud.aws;
 
+import java.util.Locale;
+import java.util.function.Function;
+
 import com.amazonaws.ClientConfiguration;
 import com.amazonaws.Protocol;
 import com.amazonaws.services.s3.AmazonS3;
 import org.elasticsearch.common.component.LifecycleComponent;
-import org.elasticsearch.common.settings.SecureSetting;
 import org.elasticsearch.common.settings.SecureString;
 import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Setting.Property;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.TimeValue;
 
-import java.util.Locale;
-import java.util.function.Function;
-
 public interface AwsS3Service extends LifecycleComponent {
 
-    // Global AWS settings (shared between discovery-ec2 and repository-s3)
+    // Legacy global AWS settings (shared between discovery-ec2 and repository-s3)
     // Each setting starting with `cloud.aws` also exists in discovery-ec2 project. Don't forget to update
     // the code there if you change anything here.
     /**
      * cloud.aws.access_key: AWS Access key. Shared with discovery-ec2 plugin
      */
-    SecureSetting<SecureString> KEY_SETTING = SecureSetting.secureString("cloud.aws.access_key", null, true, Property.Shared);
-
+    Setting<SecureString> LEGACY_KEY_SETTING = new Setting<>("cloud.aws.access_key", "", SecureString::new,
```

Comment:
I don't think we have use java @Deprecated for `Setting`s anywhere else, and I'm not sure what advantage it would bring.  It's completely internal, and the deprecated property signals to the user (through automatic deprecation logging on use) that the setting is going away.